sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParametersCorrectOrder.cs

```diff
@@ -149,6 +148,14 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     additionalLocations: secondaryLocations,
                     messageArgs: methodSymbol.Name));
             }
+
+            bool IdentifierWithSameNameAndTypeExists(IParameterSymbol parameter) =>
+                identifierArguments.Any(ia =>
+                    ia.IdentifierName == parameter.Name &&
+                    GetTypeSymbol(ia.ArgumentSyntax.Expression).DerivesOrImplements(parameter.Type));

```

Comment:
Could _GetTypeSymbol_ ever return null?

Reply:
It could, but `DerivesOrImplements` returns false when its argument is null (it is an extension method)